src/main/java/com/tsmms/skoop/user/command/UserPermissionCommandService.java

```diff
@@ -39,23 +55,30 @@ public UserPermissionCommandService(UserRepository userRepository, UserPermissio
 				.searchParamsMap(new String[]{"id", command.getOwnerId()})
 				.build());
 
-		// Remove all existing user permissions for the owner.
-		userPermissionRepository.deleteByOwnerId(command.getOwnerId());
-
 		// Create new user permissions for the given scopes and authorized users.
-		List<UserPermission> userPermissions = new ArrayList<>();
+		final List<UserPermission> userPermissions = new LinkedList<>();
 		scopeAuthorizedUsers.forEach((scope, authorizedUserIds) -> {
-			List<User> authorizedUsers = new ArrayList<>();
-			userRepository.findAllById(authorizedUserIds).forEach(authorizedUsers::add);
+			// Remove all existing user permissions for the owner.
+			userPermissionRepository.deleteByOwnerIdAndScope(command.getOwnerId(), scope);
+			final List<User> authorizedUsers = new LinkedList<>();
+			if (allUsersAuthorized.get(scope)) {
+				stream(userRepository.findAll().spliterator(), false)
```

Comment:
This logic is problematic because it only authorizes only all those users which are known to the system at the time of execution. What if a new user signs up at a later point of time? That user will not be authorized unless the owner updates the authorization.

Reply:
You are right... Thanks. I missed that.